File: projects/telegram/java/service/user/UserServiceImpl.java
package jk.uz.pdp.service.user;

import jk.uz.pdp.model.User;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

public class UserServiceImpl implements UserService {
    private Long id = 0L;
    @Override
    public int create(User user) {
        user.setId(id++);
        users.add(user);
        return 1;
    }

    @Override
    public User getById(Long id) {
        for (User u:users) {
            if(u.getId().equals(id)) {
                return u;
            }
        }
        return null;
    }

    @Override
    public User updateById(User update) {
        int i = 0;
        for (User u:users) {
            if(u.getId().equals(update.getId())) {
                users.set(i, update);
                return update;
            }
            i++;
        }
        return null;
    }

    @Override
    public int deleteById(Long id) {
        for (User u:users) {
            if(u.getId().equals(id)) {
                users.remove(u);
                break;
            }
        }
        return 1;
    }

    @Override
    public int setSMSCode(int smsCode, String phoneNumber) {
        for (User u: users) {
            if(u.getPhoneNumber().equals(phoneNumber)) {
                u.setSmsCode(smsCode);
            }
        }
        return 0;
    }

    @Override
    public User signIn(String phoneNumber, int smsCode) {
        for(User u:users) {
            if(u.getPhoneNumber().equals(phoneNumber) && u.getSmsCode() == smsCode) {
                return u;
            }
        }
        return null;
    }

    public ArrayList<User> getUsersByIds(Set<Long> ids) {
        ArrayList<User> userList = new ArrayList<>();

        for (User u:users) {
            if(ids.contains(u.getId())) {
                userList.add(u);
            }
        }
        return userList;
    }

    @Override
    public ArrayList<User> searchByName(String partOfName ) {
        ArrayList<User> foundUsers = new ArrayList<>();
        for(User user:users){
            if (user.getFullName().toLowerCase().contains(partOfName.toLowerCase())) {
                foundUsers.add(user);
            }
        }
        return foundUsers;
    }
}

File: projects/telegram/service/user/user-service-impl.ts
import { User } from "../../entities/user";
import { UserService } from "./user-service";

export class UserServiceImpl implements UserService {
	users: User[];
	private id = 0;

	create(user: User): number {
		user.setId(this.id++);
		this.users.push(user);
		return 1;
	}

	getById(id: number) {
		for (let user of this.users) {
			if (user.getId() === id) {
				return user;
			}
		}

		return null;
	}

	updateById(update: User) {
		for (let idx in this.users) {
			const user = this.users[idx];

			if (user.getId() === update.getId()) {
				this.users[idx] = update;
				return update;
			}
		}
		return null;
	}

	deleteById(id: number) {
		for (let idx in this.users) {
			const user = this.users[idx];

			if (user.getId() === id) {
				this.users.splice(+idx, 1);
				return 1;
			}
		}

		return 0;
	}

	setSMSCode(smsCode: number, phoneNumber: string) {
		for (let user of this.users) {
			if (user.phoneNumber === phoneNumber) {
				user.smsCode = smsCode;
				return 1;
			}
		}

		return 0;
	}

	signIn(phoneNumber: string, smsCode: number): User {
		for (let user of this.users) {
			if (user.phoneNumber === phoneNumber) {
				if (user.smsCode === smsCode) return user;
				else throw new Error("Wrong smsCode");
			}
		}

		throw new Error("Wrong phone number or sms code");
	}

	searchByName(name: string): User[] {
		const foundUsers: User[] = [];

		for (let user of this.users) {
			if (user.fullName?.toLowerCase().includes(name.toLowerCase())) foundUsers.push(user);
		}

		return foundUsers;
	}

	getUsersByIds(ids: number[]): User[] {
		const users: User[] = [];
		for (let user of this.users) {
			if (ids.includes(user.getId())) users.push(user);
		}
		return users;
	}
}
